cypress/support/commands/api_commands.js

```diff
@@ -114,12 +131,17 @@ Cypress.Commands.add(
       // Check if this request is already registered
       const isAlreadyRegistered = !!interceptedAliasesMap[aliasObjectKey];
       // Setting wasRequestIntercepted flag to false initially
-      setRequestIntercepted(false);
+      if (waitOnlyIfRequestIntercepted) {
+        setRequestIntercepted(false);
```

Comment:
As an aside, what is the default for `getRequestIntercepted` if you don't set this `false` here?  My gut is saying we should default `getRequestIntercepted` to false even if the test didn't call `setRequestIntercepted(false)`.

Note, I'm not saying change this line.  I think it makes sense to be explicit here.  I'm just curious if there could be a difficult to debug issue if we don't default it to false.